sycl/tools/sycl-ls/sycl-ls.cpp

```diff
@@ -106,6 +106,12 @@ int main(int argc, char **argv) {
     return EXIT_FAILURE;
   }
 
+  if (getenv("SYCL_DEVICE_FILTER")) {
+    std::cout
+        << "Warning: SYCL_DEVICE_FILTER environment variable is set. Unset it "
+           "to get full list of devices and correct device indexes."
+        << std::endl;
+  }
```

Comment:
This is new feature request not connected to the bug I am fixing. I suggest to create issue and have broad discussion to make sure that all are on the same change before do major changes in interfaces.

Reply:
I strongly recommend putting the device number when listing the devices in this PR since your code changes in this PR already assumes this change in sycl-ls.
That's much clearer for users what device number to use for SYCL_DEVICE_FILTER.
I request @romanovvlad to comment on this.
If he is okay to delay to another PR, I am okay too.

Reply:
@romanovvlad, friendly ping on question from @bso-intel